docs/content/using/using_hostdefinition.md

```diff
@@ -0,0 +1,72 @@
+
+The `HostDefiner` identifies the hosts available for host definition on each storage system and controls each of the host definitions. To see the statuses of the managed host definitions by the host definer, use the `hostdefinition` command.
+
+Use the `hostdefinitions` command to see the phase status of all definitions on the storage side.
+
+     $> kubectl get hostdefinitions
+
+|Phase|Description|
+|---------|--------|
+|Ready|Host definition is created and is working as expected.|
+|PendingForCreation|Host definition did not complete during the last attempt. The `HostDefiner` will try again.|
+|PendingForDeletion|Host deletion did not complete during the last attempt. The `HostDefiner` will try again.|
+|Error|Host definition or deletion did not complete and will not try again.|
+
+## Recovering from an Error state
+
+If any of the host definitions have an Error status, follow this procedure to have the `HostDefiner` reattempt to define the hosts.
+
+1. Undeploy the CSI node pod from the relevant node that the `HostDefinition` is a part of.
+2. Verify that all `HostDefinition` instances of the node are deleted.
+     
+          $> kubectl get hostdefinition | grep <hostname> | wc -l
+     
+     The output should be `0`.
+3. Redeploy the CSI node pod on the relevant node.
+
+     The HostDefiner handles all of the new host definitions.
+        
+4. Verify that the `hostdefinition` is in the _Ready_ phase.
+
+    ```
+    $> kubectl get hostdefinition
+    NAME                     AGE    PHASE   STORAGE          NODE
+    <host_definition_name1>  102m   Ready   <storage_name>   <node_name1>
+    <host_definition_name2>  102m   Ready   <storage_name>   <node_name2>
+    ```
+
+## Updating the `HostDefiner.yaml`
+
+Some of the parameters within the `HostDefiner.yaml` are configurable. Use this information to help decide whether the parameters for your storage system need to be updated.
+    
+|Field|Description|
+|---------|--------|
+|prefix|Adds a prefix to the hosts defined by the CSI driver.|
+|connectivity|Selects the connectivity type for the host ports.<br>Possible input values are:<br>- `nvme` for use with NVME over Fibre Channel connectivity<br>- `fc` for use with Fibre Channel over SCSI connectivity<br>- `iscsi` for use with iSCSI connectivity<br><br>By default, this field is blank and the driver selects the first of available connectivity types available on the storage system, according to the following hierarchy: NVMe, FC, iSCSI.|
+|allowDelete|Defines whether the `HostDefiner` is allowed to delete host definitions.<br>Input values are `true` or `false`.<br>The default value is `true`.|
+|dynamicNodeLabeling|Defines whether the `HostDefiner` chooses which nodes to manage dynamically by their CSI node resource or if the user must create the `hostdefiner.block.csi.ibm.com/managed-by=true` label on each relevant node.<br>Input values are `true` or `false`.<br>The default value is `false`, where the user must create each label.|
```

Comment:
```suggestion
|dynamicNodeLabeling|Defines whether the `HostDefiner` chooses which nodes to manage dynamically or if the user must create the `hostdefiner.block.csi.ibm.com/managed-by=true` label on each relevant node.<br>Input values are `true` or `false`.<br>The default value is `false`, where the user must create each label.|
```
`by their CSINode resource` is **how** the definer manages nodes, which is true regardless of the `dynamicNodeLabeling` value

Reply:
@oriyarde so what is the purpose of the user manually placing this label?

Reply:
updated

Reply:
opening a separate conversation for current revision